Library/ViewModels/CommentsViewModel.swift

```diff
@@ -106,43 +145,31 @@ public final class CommentsViewModel: CommentsViewModelType,
       }
     )
 
-    let commentsAndProject = Signal.combineLatest(comments, initialProject)
+    let optimisticPostComment = Signal.combineLatest(
+      initialProject,
+      currentUser.skipNil()
+    ).takePairWhen(self.postCommentButtonTappedProperty.signal.skipNil())
+      .map(unpack)
+      .map(Comment.optimisticComment)
+
+    self.postCommentSubmitted = optimisticPostComment.ignoreValues()
+
+    let commentsWithOptimisticPostComment = Signal
```

Comment:
Shorter version:

```swift
let commentsWithOptimisticPostComment = Signal.combineLatest(comments, optimisticPostComment)
  .map { loadedComments, optimisticComment in
    [optimisticComment] + loadedComments
  }
```

Reply:
This approach transforms the variable to a function `(Array<Comment>, Array<Comment>.ArrayLiteralElement) -> [Comment]` instead of the return type specified

Reply:
Oh weird, yeah I see one needs to do this to make it work:

```swift
let commentsWithOptimisticPostComment = Signal
  .combineLatest(comments, optimisticPostComment)
  .map { loadedComments, optimisticComment in
    ([optimisticComment] + loadedComments)
  }
  .map { $0 as Array }
```

Anyway, up to you, either is fine.